manticore/ethereum.py

```diff
@@ -668,44 +688,47 @@ def _compile(source_code, contract_name):
             :param contract_name: a string with the name of the contract to analyze
             :return: name, source_code, bytecode, srcmap, srcmap_runtime, hashes
         """
-        solc = "solc"
-        with tempfile.NamedTemporaryFile() as temp:
-            temp.write(source_code)
-            temp.flush()
-            p = Popen([solc, '--combined-json', 'abi,srcmap,srcmap-runtime,bin,hashes,bin-runtime', '--allow-paths', '.', temp.name], stdout=PIPE, stderr=PIPE)
+        if isinstance(source_code, str):
```

Comment:
Change function docstring to say that `source_code` is either `str` or a file object

Reply:
fixed in https://github.com/trailofbits/manticore/pull/873/commits/6cb4bfaf1ceded779c20349d5311cff4b9d9675d
I also updated the doc string for the tuple of return values